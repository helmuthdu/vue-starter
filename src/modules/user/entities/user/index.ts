export * from './user.type';
export * from './user.model';
