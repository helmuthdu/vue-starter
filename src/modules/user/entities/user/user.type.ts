export type UserSchema = {
  email: string;
  emailVerified?: boolean;
  id?: number;
  password?: string;
  phoneNumber?: string;
  phoneNumberVerified?: boolean;
  newsletter?: boolean;
  token?: string;
  userName?: string;
  googleId?: string;
  facebookId?: string;
};
