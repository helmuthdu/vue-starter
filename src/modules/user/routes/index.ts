import { paths } from '@/modules/user/routes/paths';
import { RouteRecordRaw } from 'vue-router';

export const routes: Array<RouteRecordRaw> = [
  {
    path: paths.root,
    component: () => import('../layouts/user.layout.vue'),
    children: [
      {
        path: paths.signIn.path,
        name: paths.signIn.name,
        component: () => import('./sign-in/sign-in.route.vue'),
      },
    ],
  },
];
