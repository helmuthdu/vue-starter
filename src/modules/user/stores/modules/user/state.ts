export interface State {
  username: string;
  email: string;
  isLoggedIn: boolean;
  token: string;
}

export const initialState = {
  username: '',
  email: '',
  isLoggedIn: false,
  token: ''
};

export const name = 'user';

export const state = (): State => JSON.parse(JSON.stringify(initialState));
