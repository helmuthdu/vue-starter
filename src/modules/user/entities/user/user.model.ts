import { UserSchema } from './user.type';

export class User implements UserSchema {
  readonly email: string;
  readonly emailVerified: boolean;
  readonly id?: number;
  readonly password: string;
  readonly phoneNumber?: string;
  readonly phoneNumberVerified: boolean;
  readonly userName: string;
  readonly googleId?: string;
  readonly facebookId?: string;
  readonly token?: string;

  static create(user?: UserSchema) {
    return Object.assign({}, new User(), { ...(user ?? {}) }) as UserSchema;
  }
}
