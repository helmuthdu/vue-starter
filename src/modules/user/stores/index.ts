import * as user from './user.store';

export { user };
