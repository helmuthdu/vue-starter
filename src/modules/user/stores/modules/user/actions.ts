import { State as AppState } from '../../..';
import { userApi, AuthenticatePayload } from '../../../api/user.api';
import { ActionContext, ActionTree } from 'vuex';
import { State } from './state';
import { UserActionTypes } from './types';

export interface Actions<S, R> extends ActionTree<S, R> {
  [UserActionTypes.SIGN_IN]: (context: ActionContext<S, R>, payload: any) => void;
  [UserActionTypes.SIGN_OUT]: (context: ActionContext<S, R>) => void;
}

export const actions: Actions<State, AppState> = {
  async [UserActionTypes.SIGN_IN]({ commit }, payload: AuthenticatePayload) {
    commit(UserActionTypes.SET_USER, {
      ...(await userApi.get(payload)),
      isLoggedIn: true
    });
  },
  [UserActionTypes.SIGN_OUT]({ commit }) {
    commit(UserActionTypes.DEL_USER);
  }
};
