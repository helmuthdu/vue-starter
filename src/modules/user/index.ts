import { routes } from './routes';
import { paths } from './routes/paths';
import * as store from './stores';

export type State = Readonly<{
  user: store.user.State;
}>;

const stores = {
  user: store.user.store
};

export { paths, routes, stores };
