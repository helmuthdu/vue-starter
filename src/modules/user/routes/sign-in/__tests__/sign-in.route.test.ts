import { mount, VueWrapper } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';
import { State, UserActionTypes } from '@/modules/user/stores/modules/user';
import SignInRoute from '../sign-in.route.vue';

describe('Auth/Route -> SignIn', () => {
  let store: Store<State>;
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    store = new Vuex.Store({
      actions: {
        [UserActionTypes.SIGN_IN]: jest.fn()
      }
    });
    wrapper = mount(SignInRoute, { global: { plugins: [store] } });
  });

  it('should renders with props', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
