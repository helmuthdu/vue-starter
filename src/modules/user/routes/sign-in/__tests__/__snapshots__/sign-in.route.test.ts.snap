// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth/Route -> SignIn should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div>
        <form>
          <input
            label="Email"
            required=""
          />
          <input
            label="Password"
            required=""
            type="password"
          />
          <button
            color="pink"
            dark=""
          >
            Sign-in
          </button>
        </form>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$store": Store {
            "_actionSubscribers": Array [],
            "_actions": Object {
              "USER/SIGN_IN": Array [
                [Function],
              ],
            },
            "_committing": false,
            "_makeLocalGettersCache": Object {},
            "_modules": ModuleCollection {
              "root": Module {
                "_children": Object {},
                "_rawModule": Object {
                  "actions": Object {
                    "USER/SIGN_IN": [MockFunction],
                  },
                },
                "context": Object {
                  "commit": [Function],
                  "dispatch": [Function],
                },
                "runtime": false,
                "state": Object {},
              },
            },
            "_modulesNamespaceMap": Object {},
            "_mutations": Object {},
            "_state": Object {
              "data": Object {},
            },
            "_subscribers": Array [],
            "_wrappedGetters": Object {},
            "commit": [Function],
            "dispatch": [Function],
            "getters": Object {},
            "strict": false,
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        "store": Store {
          "_actionSubscribers": Array [],
          "_actions": Object {
            "USER/SIGN_IN": Array [
              [Function],
            ],
          },
          "_committing": false,
          "_makeLocalGettersCache": Object {},
          "_modules": ModuleCollection {
            "root": Module {
              "_children": Object {},
              "_rawModule": Object {
                "actions": Object {
                  "USER/SIGN_IN": [MockFunction],
                },
              },
              "context": Object {
                "commit": [Function],
                "dispatch": [Function],
              },
              "runtime": false,
              "state": Object {},
            },
          },
          "_modulesNamespaceMap": Object {},
          "_mutations": Object {},
          "_state": Object {
            "data": Object {},
          },
          "_subscribers": Array [],
          "_wrappedGetters": Object {},
          "commit": [Function],
          "dispatch": [Function],
          "getters": Object {},
          "strict": false,
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$store": Store {
          "_actionSubscribers": Array [],
          "_actions": Object {
            "USER/SIGN_IN": Array [
              [Function],
            ],
          },
          "_committing": false,
          "_makeLocalGettersCache": Object {},
          "_modules": ModuleCollection {
            "root": Module {
              "_children": Object {},
              "_rawModule": Object {
                "actions": Object {
                  "USER/SIGN_IN": [MockFunction],
                },
              },
              "context": Object {
                "commit": [Function],
                "dispatch": [Function],
              },
              "runtime": false,
              "state": Object {},
            },
          },
          "_modulesNamespaceMap": Object {},
          "_mutations": Object {},
          "_state": Object {
            "data": Object {},
          },
          "_subscribers": Array [],
          "_wrappedGetters": Object {},
          "commit": [Function],
          "dispatch": [Function],
          "getters": Object {},
          "strict": false,
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "email": undefined,
    "hasOwnProperty": [Function],
    "password": undefined,
    "signIn": [Function],
    "submit": [Function],
  },
  "rootVM": Object {},
  "wrapperElement": <div>
    <form>
      <input
        label="Email"
        required=""
      />
      <input
        label="Password"
        required=""
        type="password"
      />
      <button
        color="pink"
        dark=""
      >
        Sign-in
      </button>
    </form>
  </div>,
}
`;
